### PR TITLE
Update G1/G2 Required Test validation logic, add test to verify SED End Date

### PR DESCRIPTION
**Pull request by ashwinimore**

### CODE REVIEW

File: src/test/resources/Features/OCD-1970-ModificationOfG1G2Values.feature [hunk 1]
```diff
@@ -6,34 +6,28 @@ Scenario Outline: Mapping legacy values for G1/G2
 	When I open details for criteria "<CRITERIA>"
 	Then Measure Successfully Tested for "<gOption>" should display "<Measure_Name>" for "<CRITERIA>" 
 	When I look at  Successfully Tested G1G2 Measures
-	Then the "<Measure_Name>" shows Required Text as "<Required_Text>" and associated criteria as "<CRITERIA>"
+	Then the Required Test shows as "<Required_Test>" for "<Measure_Name>" and associated criteria "<CRITERIA>"
 	Examples:
-    |DB_ID|CRITERIA|gOption|Measure_Name|Required_Text|
+    |DB_ID|CRITERIA|gOption|Measure_Name|Required_Test|
```
Comment (andlar) [inline src/test/resources/Features/OCD-1970-ModificationOfG1G2Values.feature:11]: Think `_Test` should be `_Text`

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ManageDevelopersAndProductsSteps.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+     */
+    public ManageDevelopersAndProductsSteps() {
+        driver = Hooks.getDriver();
+        if (StringUtils.isEmpty(url)) {
+            url = "http://localhost:3000/";
+           }
+    }
+
+    /**
+     * Loads a listing for given Database ID.
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ManageDevelopersAndProductsSteps.java:34]: Recommend adding a section here that explains what you're waiting for in the "ExpectedCondition" section. I think that'll be helpful for reviewers and those writing new Steps

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ListingDetailsSteps.java [hunk 3]
```diff
@@ -266,14 +266,16 @@ public void iLookAtG1G2Measures() {
     /**
      * Asserts Measure required text and associated criteria show correct.
      * @param mName expected text
-     * @param reqText is expected toggle tip text
+     * @param reqTest is expected toggle tip text
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ListingDetailsSteps.java:269]: This is "text", right? Not "test"?

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ListingDetailsSteps.java [hunk 4]
```diff
@@ -266,14 +266,16 @@ public void iLookAtG1G2Measures() {
...
-    public void assetRequiredText(final String mName, final String reqText, final String criteria) {
-        String actualString = ListingDetailsPage.requiredText(driver, mName).getText();
-        assertTrue(actualString.contains(reqText), "Expect \"" + reqText + "\" to be found in \"" + actualString + "\"");
+    @Then("^the Required Test shows as \"([^\"]*)\" for \"([^\"]*)\" and associated criteria \"([^\"]*)\"$")
+    public void requiredTestValidationBasedOnMeasureName(final String reqTest, final String mName, final String criteria) {
+        String actualString = ListingDetailsPage.measureNameForReqTest(driver, reqTest).getText();
+        System.out.println(actualString);
+        assertTrue(actualString.contains(mName), "Expect \"" + mName + "\" to be found in \"" + actualString + "\"");
         String associatedCriteria = ListingDetailsPage.associatedCriteria(driver, mName).getText();
+        System.out.println(associatedCriteria);
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ListingDetailsSteps.java:278]: Looks like a debugging line. Probably not needed any more, right?

File: src/test/java/gov/healthit/chpl/aqa/pageObjects/ManageDevelopersAndProductsPage.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            return element;
+        }
+
+        /**
+         * Returns element that holds date string.
+         * @param driver WebDriver
+         * @return date element
+         */
+        public static WebElement sedEndDateOfTesting(final WebDriver driver) {
+            element = driver.findElement(By.xpath("//*[@id=\"main-content\"]/section/div/div[3]/div[5]/div/div/div/div[2]/span/text()[16]"));
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/pageObjects/ManageDevelopersAndProductsPage.java:31]: Wow. We definitely need some IDs on this page. Maybe one for each "Section" to start with? (Developer, Product, Version, Listing?)

File: src/test/java/gov/healthit/chpl/aqa/pageObjects/ListingDetailsPage.java [hunk 6]
```diff
@@ -217,17 +217,17 @@ public static WebElement nosedData(final WebDriver driver) {
     /**
      * Returns the element that holds Required Text for a Measure Name in Successfully Tested G1/G2 Measures section .
      * @param driver WebDriver
-     * @param mName is measure Name
+     * @param reqTest is Required Test text
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/pageObjects/ListingDetailsPage.java:220]: Wait, is it "Test"? Or is it "Text"? Now I have no idea

**Review by andlar — APPROVED** [review]
Lots of cosmetic things, but nothing that would hold up a merge. Recommend merging this in now, and making some (or all) of the changes I brought up later and rolling those into whatever your next PR is

**Review by kekey1 — APPROVED** [review]

File: src/test/resources/Features/OCD-1970-ModificationOfG1G2Values.feature (hunk 1 above)
Comment (ashwinimore) [reply src/test/resources/Features/OCD-1970-ModificationOfG1G2Values.feature:11]: The field name is 'Required Test' , so i changed it to have the field name

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ListingDetailsSteps.java (hunk 4 above)
Comment (ashwinimore) [reply src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ListingDetailsSteps.java:278]: yes, forgot to remove it. I took it out, change will be in next PR

**Review by brianlindsy — APPROVED** [review]

**Pull request merged**